### PR TITLE
chore: refactor: extract ClusterConfig

### PR DESCRIPTION
**Pull request by AlekSi**
Extract ClusterConfig and related types.
Make one huge file a bit smaller.

No real changes in this PR, just code movement and comment tweaks.

`MachineConfig` and co will follow.

- [ ] you linked an issue (if applicable)
- [ ] you included tests (if applicable)
- [x] you ran conformance (`make conformance`)
- [x] you formatted your code (`make fmt`)
- [x] you linted your code (`make lint`)
- [x] you generated documentation (`make docs`)
- [x] you ran unit-tests (`make unit-tests`)

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -164,9 +164,11 @@ issues:
     - path: internal/app/machined/pkg/system/services
       linters:
         - dupl
-    - path: pkg/config/types/v1alpha1
```
Comment (AlekSi) [inline .golangci.yml:167]: This path no longer exist

File: .golangci.yml [hunk 2]
```diff
@@ -164,9 +164,11 @@ issues:
...
+
+    # pkg/machinery module
+    - path: config/types/v1alpha1
```
Comment (AlekSi) [inline .golangci.yml:169]: `pkg/machinery/config/types/v1alpha1` does not work – golangci-lint uses paths within the module

File: pkg/machinery/config/types/v1alpha1/v1alpha1_etcdconfig.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+// CA implements the config.Etcd interface.
+func (e *EtcdConfig) CA() *x509.PEMEncodedCertificateAndKey {
+	return e.RootCA
+}
+
+// ExtraArgs implements the config.Etcd interface.
+func (e *EtcdConfig) ExtraArgs() map[string]string {
+	if e.EtcdExtraArgs == nil {
+		e.EtcdExtraArgs = make(map[string]string)
```
Comment (smira) [inline pkg/machinery/config/types/v1alpha1/v1alpha1_etcdconfig.go:40]: probably not for this PR, but we should review places like this one (writing to config object) and remove them, as it's a perfect data race

**Review by smira — APPROVED** [review]

**Pull request merged**